src/libostree/ostree-repo-commit.c

```diff
@@ -770,7 +770,7 @@ write_object (OstreeRepo         *self,
     }
 
   if (!checksum_input)
-    actual_checksum = expected_checksum;
+    actual_checksum = g_strdup (expected_checksum);
```

Comment:
I know it seems minor but we malloc a whole lot right now for each object, and I'd like to reduce that.  The pattern I've tended to use here is...well, let me push an alternative patch: